### PR TITLE
Optimize Compositor by combining updates

### PR DESCRIPTION
**Pull request by willmcgugan**
- Deferred rendering of widget updates. When a widget update is request, Textual will wait for 50ms for further updates.
- The compositor can now render multiple updates in a single render.

### CODE REVIEW

File: src/textual/_compositor.py [hunk 1]
```diff
@@ -116,6 +136,43 @@ def __init__(self) -> None:
         # The points in each line where the line bisects the left and right edges of the widget
         self._cuts: list[list[int]] | None = None
 
+    @classmethod
```
Comment (willmcgugan) [inline src/textual/_compositor.py:139]: Adapted from @darrenburns 's region_groups.py

File: src/textual/_compositor.py [hunk 2]
```diff
@@ -91,6 +92,33 @@ def __rich_repr__(self) -> rich.repr.Result:
         yield "height", height
 
 
+@rich.repr.auto
+class SpansUpdate:
+    """A renderable that apples updated spans to the screen."""
```
Comment (DrBenton) [inline src/textual/_compositor.py:97]: typo :-) :apple:

File: src/textual/_compositor.py [hunk 3]
```diff
@@ -116,6 +144,42 @@ def __init__(self) -> None:
         # The points in each line where the line bisects the left and right edges of the widget
         self._cuts: list[list[int]] | None = None
 
+    @classmethod
+    def _regions_to_spans(
+        cls, regions: Iterable[Region]
+    ) -> Iterable[tuple[int, int, int]]:
+        """Converts the regions to horizontal spans. Spans will be combined if the overlap
```
Comment (DrBenton) [inline src/textual/_compositor.py:151]: should be _"if they overlap"_, I guess?

**Review by DrBenton — APPROVED** [review]
These geometrical and time-based batching operations are far too clever for my little brain :brain: (especially with the quite little knowledge I have of the Textual internals at the moment), so I can only comment on things that really don't matter, like typos in comments - sorry :grimacing:

**Pull request merged**